soroban-env-host/src/auth.rs

```diff
@@ -788,6 +788,10 @@ impl AuthorizationManager {
             AuthorizationMode::Enforcing => self.require_auth_enforcing(host, address, &function),
             // metering: free for recording
             AuthorizationMode::Recording(recording_info) => {
+                // First try to find the tracker for this exact address object.
```

Comment:
What is the meaning of "exact"? What's the distinction between an exactly the same, and a matching address object?

Reply:
Exactly the same == exactly the same `Val` (i.e. the same host object). Matching == same value. Same `Val` takes precedence to help with the cases where `Address` is passed around (which should be probably majority of the cases). This only matters for complex cases where multiple addresses are involved and duplicates are possible.

Reply:
Got it, exact is same handle, and matching is same address held in a different host object, because the values aren't interned? Got it.